data/api/client-server/report_content.yaml

```diff
@@ -29,7 +92,7 @@ paths:
         will require the homeserver to check whether a user is joined to
         the room. To combat this, homeserver implementations should add
         a random delay when generating a response.
-      operationId: reportContent
+      operationId: reportEvent
```

Comment:
I'm unsure what consequences changing this has but the old ID felt unsuitable now that there are two endpoints in the module.

Reply:
Looks like it's used to generate bindings, and as such we probably shouldn't change it...
https://github.com/matrix-org/matrix-spec/blob/2cbf6067a6d50ea2160e72d2a570f2cb8eeb0a20/meta/documentation_style.rst?plain=1#L113-L114

@turt2live curious if you happen to know more about this?

Reply:
I suppose if we want to be safe keeping `operationId: reportContent` on the existing endpoint and `operationId: reportRoom` on the new one would not be the end of the world. So maybe reverting back would be best?

Reply:
Indeed, `operationId` usually ends up as the method or class name in generated APIs, and therefore, may (actually will, in the vast majority of cases) break the generated code. However, given that even more fine-grained changes (e.g. adding a field in a request body) that are API-compatible in JSON may be breaking in the generated code (depending on the target language and/or the shape of the API) authors producing auto-generated code have to be careful with any upgrades of the API definitions. In particular, 1.12 has already introduced breaking changes elsewhere from the Quotient perspective. So I wouldn't be too concerned around it, just be cognisant that not everything back-compatible in JSON remains such in a native API.

Reply:
We concluded in #matrix-docs:matrix.org that changing the operation ID is not ideal but acceptable in this case as otherwise it'll be even harder to fix in future.